### DISCUSSION TITLE
Is it possible to defined a filled circle/ellipse?

### DISCUSSION THREAD

**Discussion by johnthagen** [mozman/ezdxf#1161] [Q&A] [answered]
When I create a regular circle using `add_circle()` and view it in a DXF viewer:

<img width="164" alt="Screenshot 2024-09-11 at 2 59 07 PM" src="https://github.com/user-attachments/assets/7bfde6f4-2203-490f-8cb2-9ee42d729ad0">

It is not filled. I've created `lwpolyline`s and `hatch`s that are filled and I'd like to define regular circles as filled as well.

Does DXF support something like this?

**Comment by mozman** [comment, marked as answer]
You have to create a `HATCH` with an `EdgePath` as boundary.

See `HATCH` tutorial, 3rd example: https://ezdxf.mozman.at/docs/tutorials/hatch.html

```Python
import ezdxf

doc = ezdxf.new("R2000")
msp = doc.modelspace()
msp.add_ellipse((0, 0), major_axis=(0, 10), ratio=0.5)

hatch = msp.add_hatch(color=2)
edge_path = hatch.paths.add_edge_path()

# each edge path can contain line, arc, ellipse and spline elements
# important: major axis >= minor axis (ratio <= 1.)
edge_path.add_ellipse((0, 0), major_axis=(0, 10), ratio=0.5)

doc.saveas("solid_hatch_ellipse.dxf")
```

- line `EdgePath.add_line()`
- arc `EdgePath.add_arc()`
- ellipse `EdgePath.add_ellipse()`
- spline `EdgePath.add_spline()`
